### DISCUSSION TITLE
Actividad 3 - Grupo 1

### DISCUSSION THREAD

**Discussion by Fedebravo12** [frlp-utn-ingsoft/catedra#126] [Show and tell]
https://github.com/Fedebravo12/vetsoft

**Comment by ivanduva** [comment]
Buenas!
La actividad en general tiene muchos puntos altos, además de que funciona bien, pasan los tests y en general hay un buen trabajo en los PRs.
Pero vimos algunos problemas que nos parece que deberían arreglar para aprobarla:

- Al agregar la raza de la mascota, no se agrega un test e2e para probar específicamente eso, sino que se modifican otros. Habría que agregar al menos uno.
- En este PR: https://github.com/Fedebravo12/vetsoft/pull/11 no se tuvieron en cuenta los comentarios que se hicieron en la review. Esto puede pasar, pero no vemos ninguna respuesta que justifique por qué no se tuvieron en cuenta. Además, eran comentarios atinados. Habría que hacer otro PR que revise la misma persona para agregar los cambios que se pedían.

Después les hacemos algunos comentarios menores para que tengan en cuenta en las próximas entregas:

- El test test_update_pet_with_error tiene nombre duplicado y ambos tienen muchos asserts para ser de unidad.
- Los tests test_validation_invalid_speciality y test_validation_invalid_speciality_none no hacen nada.
- Este commit tiene un mal mensaje: https://github.com/Fedebravo12/vetsoft/pull/12/commits/934e64309a768cf153bfe78abbc02fb069e3af22. No es necesario agregar cuestiones relativas al código en el mensaje, es redundante porque esa info ya la tiene el commit. El mensaje debería ser una síntesis de la funcionalidad que agrega el commit.
- Vimos que en un par de PRs, otra persona que no fue el que lo creó realizó el merge con main. No le esquiven a la resolución de conflictos, si pasa esto de nuevo, lo vamos a mandar a corregir.
- Tanto en la dirección del provider como en el precio del producto, permite guardar los campos vacíos al editar más allá de que no se modifica el atributo. Deberían atajarlo con un error en el form.

Después el trabajo está bastante bien, sobre todo en los tests que agregaron muchos y la mayoría con buen criterio. También vemos un buen feedback en los PRs en general, con comentarios sobre el código que se correspondían luego con commits.
Arreglando esos 2 puntos de más arriba ya estaría aprobado

Buen trabajo! 
Saludos

**Reply by germansalinas1994** [reply]
Se corrigió lo siguiente

Se agregaron tests e2e para la validacion de la raza de mascota
Se realizó una nueva pull request teniendo en cuenta los cambios sugeridos en https://github.com/Fedebravo12/vetsoft/pull/11

En cuanto a lo corregido en el test test test_update_pet_with_error, no pudimos encontrar ese error, quizas es confuso que tenga test para error de weight y para error de breed, de todas formas se realizan los test para ambos casos

El mensaje de este commit https://github.com/Fedebravo12/vetsoft/commit/934e64309a768cf153bfe78abbc02fb069e3af22 lo tenemos en cuenta para la proxima entrega a modo de casos a corregir

Se solucionaron los errores en la validación de los update de provider y product. Tuvimos una complicación en este caso donde tuvimos que revertir una pull request (cuestión que también dio el caso para probar) y ya quedo solucionado!

**Comment by ivanduva** [comment]
Perfecto, queda aprobado. Saludos!
